modules/binary.js

```diff
@@ -81,27 +81,37 @@ exports.ByteArray = ByteArray;
 exports.ByteString = ByteString;
 
 /**
- * Converts the String to a mutable ByteArray using the specified encoding.
- * @param {String} string The string to convert into a ByteArray
- * @param {String} charset the name of the string encoding. Defaults to 'UTF-8'
- * @returns {ByteArray} a ByteArray representing the string
+ * Converts a String or Binary instance to a mutable ByteArray using the specified encoding.
+ * @param {String|Binary} str The String or Binary to convert into a ByteArray
+ * @param {String} charset the string's encoding. Defaults to 'UTF-8'
+ * @returns {ByteArray} a ByteArray representing the str
  * @example const binary = require("binary");
  * const ba = binary.toByteArray("hello world");
  */
-exports.toByteArray = (string, charset) => {
-    return new ByteArray(String(string), charset || 'utf8');
+exports.toByteArray = (str, charset) => {
```

Comment:
Imho both methods should throw an error if str is null or undefined - otherwise you'd end up with the string "null" or "undefined" converted into a ByteArray/ByteString.

Reply:
Added a stricter check in bf7c1fb